### PR TITLE
feat(argocd-apps): Generate application, applicationset, project with template

### PR DESCRIPTION
**Pull request by joelee2012**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by jmeridth** [comment]
Please update the annotations for artifacthub at the [bottom of the Chart.yaml file](https://github.com/argoproj/argo-helm/pull/2025/files#diff-70940c1fa6e9639f25dffbcb6c7fa9110565e106901d441332ad1918c2dd19ddR15-R18).  Thank you.

File: charts/argocd-apps/values.yaml [hunk 1]
```diff
@@ -137,7 +177,8 @@ applicationsets: []
 # @default -- `[]` (See [values.yaml])
 ## This function in tech preview stage, do expect unstability or breaking changes in newer versions. Bump image.tag if necessary.
 ## Ref: https://github.com/argoproj-labs/argocd-extensions
-extensions: []
+extensions:
```
Comment (mkilchhofer) [inline charts/argocd-apps/values.yaml:180]: Please revert to existing style

File: charts/argocd-apps/values.yaml [hunk 2]
```diff
@@ -40,6 +40,46 @@ applications: []
 #   - name: url
 #     value: https://argoproj.github.io/
 
+applicationTemplates:
+  []
```
Comment (mkilchhofer) [inline charts/argocd-apps/values.yaml:44]: Please follow the style we currently use and put the brackets on the same line as the key:

```yaml
myList: []
myDictionary: {}
```

File: charts/argocd-apps/ci/application-templates.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+applicationTemplates:
+  - apps:
+      - name: guestbook
+        project: guestbook-project
+    template: |-
```
Comment (mkilchhofer) [inline charts/argocd-apps/ci/application-templates.yaml:5]: Does it also work with YAML/object rather than type string?

**Review by mkilchhofer — CHANGES_REQUESTED** [review]

File: charts/argocd-apps/ci/application-templates.yaml (hunk 3 above)
Comment (joelee2012) [reply charts/argocd-apps/ci/application-templates.yaml]: yes, it does

**Comment by jmeridth** [comment]
You need to update the documentation with your new `itemTemplates`

**Comment by joelee2012** [comment]
> You need to update the documentation with your new `itemTemplates`

updated

**Comment by joelee2012** [comment]
@jmeridth is it possible to enable all testing for PR by default ? so that the owner can fix every issue before requiring code review. how do you think ?

**Comment by jmeridth** [comment]
@joelee2012 tgats fine. Your update doc commit is not signed. Can't pass DCO.

**Comment by joelee2012** [comment]
@jmeridth btw, should i update version to `1.1.0` instead of `1.0.1`  as it's new feature ?

**Comment by joelee2012** [comment]
@jmeridth @mkilchhofer are you able check again ?

**Review by joelee2012 — COMMENTED** [review]
done

**Review by jmeridth — APPROVED** [review]

**Comment by joelee2012** [comment]
@mkilchhofer are you able to check again ?

File: charts/argocd-apps/templates/applications.yaml [hunk 4]
```diff
@@ -49,3 +49,19 @@ spec:
...
+{{ tpl $template (set . "Template" $.Template) }}
+{{- end }}
+{{- else }}
+{{- $template := .template | toYaml -}}
+{{- range .items }}
+---
+{{ tpl $template (set . "Template" $.Template) }}
+{{- end }}
+{{- end }}
+{{- end }}
```
Comment (mkilchhofer) [inline charts/argocd-apps/templates/applications.yaml:67]: Can you move this block to a dedicated file. Doesn't make sense to keep this inside applications.yaml.

Please create a new file for that. Eg. `item-templates.yaml`.

**Comment by joelee2012** [comment]
@mkilchhofer can you help review again ?

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**